File: backend/app/schemas/agent.py
"""Agent schemas."""
from __future__ import annotations
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel


class AgentResponse(BaseModel):
    id: UUID
    name: str
    description: str | None = None
    is_active: bool = True
    is_frozen: bool = False
    status: str = "active"
    cost_today_usd: float = 0.0
    cost_month_usd: float = 0.0
    cost_trend_pct: float = 0.0
    requests_today: int = 0
    last_event_at: datetime | None = None
    created_at: datetime


class AgentDetailResponse(AgentResponse):
    metrics: dict = {}


class AgentUpdateRequest(BaseModel):
    description: str | None = None
    is_active: bool | None = None

File: frontend/src/types/agent.ts
export type AgentStatus = 'active' | 'warning' | 'frozen' | 'inactive'

export interface Agent {
  id: string
  name: string
  description?: string | null
  is_active: boolean
  is_frozen: boolean
  status: AgentStatus
  cost_today_usd: number
  cost_month_usd: number
  cost_trend_pct: number
  requests_today: number
  last_event_at: string | null
  created_at: string
}

export interface AgentsListResponse {
  data: Agent[]
  pagination: {
    total: number
    limit: number
    offset: number
    has_more: boolean
  }
}
